pandas/core/arrays/base.py

```diff
@@ -585,7 +585,7 @@ def argmin(self):
         """
         return nargminmax(self, "argmin")
 
-    def argmax(self):
+    def argmax(self, *args):
```

Comment:
umm what args are getting passed here? we would like to be explict (and then call validation routines on these)

Reply:
`Series.argmax` calls `core.nanops.nanargmax` which calls `_values.argmax` with argument `axis`, which is `None` in this case. I think alternatively we can change `core.nanops.nanargmax` code so that if `_values` is an EA instead of nparray then don't pass any arguments, but I think this has the minimum amount of code change to fix the issue. Please also refer to my reply above. Thanks.

Reply:
Oh do you mean I should change `*args` to `axis=None`, so that it is explicit?

Reply:
yep, and then use the validation that we have in other places

Reply:
> Series.argmax calls core.nanops.nanargmax

I think the solution here should rather be to let Series.argmax *not* call `nanops.nanargmax`, but let it directly call the EA argmax method (similarly like how `Series._reduce` also dispatches to the EA method in case of EAs instead of calling the nanops function)

The nanops version is doing lots of stuff that is not needed for nullable EAs.

Reply:
Thanks both for the comments.
@jreback Just made the changes. Is this what you are looking for?

@jorisvandenbossche 
> The nanops version is doing lots of stuff that is not needed for nullable EAs.

> should rather be to let Series.argmax not call nanops.nanargmax

I agree in principle, but doing that would mean making changes in Series.argmax, argmin, idxmax and idxmin. Not sure what the best approach is. The current "quick fix" seems reasonable to (the inexperienced) me.

Reply:
It are not that big changes that are required to Series.argmax/argmin (although there are some API questions to resolve, see https://github.com/pandas-dev/pandas/issues/35178)

Also, this quick fix might work for Int64 dtypes, but for example already fails again for Float64 dtype. So IMO we should try to directly fix it properly.

Reply:
@jorisvandenbossche OK. I think I agree. Here is my attempt.